src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -331,17 +329,35 @@ class AppsController(
               complete((StatusCodes.OK, List(Headers.`Marathon-Deployment-Id`(plan.id)), DeploymentResult(plan.id, plan.version.toOffsetDateTime)))
             }
           case TaskKillingMode.Wipe =>
-            onSuccess(taskKiller.kill(appId, findToKill, wipe = true)) { instances =>
-              complete(raml.InstanceList(instances.map(_.toRaml)))
+            val killInstances = async {
+              val instances = await(taskKiller.kill(appId, findToKill, wipe = true))
+              // TODO: Does it make sense to set health for a killed task?
+              val healthStatuses = await(healthCheckManager.statuses(appId))
+              instances.map { instance =>
+                EnrichedTask(appId, instance.appTask, instance.agentInfo, healthStatuses.getOrElse(instance.instanceId, Nil))
+              }
+            }
+
+            onSuccess(killInstances) { enrichedTasks: Seq[EnrichedTask] =>
+              complete((StatusCodes.OK, raml.EnrichedTasksList(enrichedTasks.toRaml)))
             }
           case TaskKillingMode.KillWithoutWipe =>
-            onSuccess(taskKiller.kill(appId, findToKill, wipe = false)) { instances =>
-              complete(raml.InstanceList(instances.map(_.toRaml)))
+            val killInstances = async {
+              val instances = await(taskKiller.kill(appId, findToKill, wipe = false))
+              val healthStatuses = await(healthCheckManager.statuses(appId))
+              instances.map { instance =>
+                EnrichedTask(appId, instance.appTask, instance.agentInfo, healthStatuses.getOrElse(instance.instanceId, Nil))
```

Comment:
I don't have an overview right now, but my initial inclination wrt to refactoring this was to remove `EnrichedTask` and add a serializer from Instance -> Task information. Basically, the serializer drops information about the instance and presents it as if it were a task, more or less.

Maybe that's a follow up PR, not sure about the impact: instead of creating an `appinfo.EnrichedTask` in the controller (that's API layer already), you could either
- directly create a `raml.EnrichedTask` from the instance here and get rid of the conversion OR
- create a conversion from instance -> `raml.EnrichedTask` (or: `raml.Task` as per the above suggestion by @timcharper)

I'd prefer the latter – I'm OK with a follow-up as I assume it's a bigger change (~40 usages of `appInfo.EnrichedTask`)

Reply:
Well, there is already a type [`Task`](https://github.com/mesosphere/marathon/blob/master/docs/docs/rest-api/public/api/v2/types/task.raml#L62). But it's not used to I'm replacing it.